AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -988,6 +988,119 @@ - (CGRect)convertRect:(CGRect)rect toNode:(ASDisplayNode *)node
   return CGRectApplyAffineTransform(rect, flattenedTransform);
 }
 
+#pragma mark - Layout Transition
+
+- (void)transitionLayoutWithAnimation:(BOOL)animated
+{
+  [self transitionLayoutToFit:_constrainedSize animated:animated];
+}
+
+- (void)transitionLayoutToFit:(ASSizeRange)constrainedSize animated:(BOOL)animated
+{
+  [self invalidateCalculatedLayout];
+  [self measureWithSizeRange:constrainedSize]; // Generate a new layout
+  [self __transitionLayoutWithAnimation:animated];
+}
+
+- (void)__transitionLayoutWithAnimation:(BOOL)animated
+{
+  _transitionContext = [[_ASTransitionContext alloc] initWithAnimation:animated delegate:self];
+
+  [self willTransitionLayout:_transitionContext];
+  if ([_insertedSubnodes count]) {
+    for (_ASDisplayNodePosition *position in _insertedSubnodes) {
+      [self _implicitlyInsertSubnode:position.node atIndex:position.index];
+    }
+    _insertedSubnodes = @[];
```

Comment:
It's a minimal distinction, but let's just use nil for this sort of thing unless there is a functional reason that the empty array is convenient